### PR TITLE
refact: async load qml

### PR DESCRIPTION
**Pull request by 18202781743**
async load qml.
  async load meta data.
  using AppletLoader to manage Applet's callback.

### CODE REVIEW

File: panels/dock/dockpanel.cpp [hunk 1]
```diff
@@ -68,7 +68,13 @@ bool DockPanel::init()
     // connect(window(), &QQuickWindow::yChanged, this, &DockPanel::onWindowGeometryChanged);
     connect(window(), &QQuickWindow::widthChanged, this, &DockPanel::onWindowGeometryChanged);
     connect(window(), &QQuickWindow::heightChanged, this, &DockPanel::onWindowGeometryChanged);
-    QMetaObject::invokeMethod(this, &DockPanel::onWindowGeometryChanged);
+
+    QObject::connect(this, &DApplet::rootObjectChanged, this, [this]() {
+        if (rootObject()) {
+            QMetaObject::invokeMethod(this, &DockPanel::onWindowGeometryChanged);
+        }
+    });
```
Comment (asterwyx) [inline panels/dock/dockpanel.cpp:76]: 既然插件都是异步加载了，加一个callback让子插件重写

Comment (18202781743) [reply panels/dock/dockpanel.cpp:76]: callback也可以，信号跟灵活些，

File: frame/CMakeLists.txt [hunk 2]
```diff
@@ -26,6 +26,8 @@ set(PRIVATE_HEADERS
 add_library(dde-shell-frame SHARED
     ${PUBLIC_HEADERS}
     ${PRIVATE_HEADERS}
+    appletloader.h
+    appletloader.cpp
     appletitem.h
     containmentitem.h
     dstypes.cpp
```
Comment (asterwyx) [inline frame/CMakeLists.txt:33]: 考虑重命名并拆分成private header

Comment (18202781743) [reply frame/CMakeLists.txt]: 移动到shell

**Review by asterwyx — APPROVED** [review]

**Pull request merged**